Dinkers/The-Curation
# 30 adding place UI

## Summary
Closes #30 

This is the big one, sorry for the massive PR!

Basically the code here builds out the entire Place screen, and gets the MVP app in a near-complete state.

## Notable changes
Honestly, the majority of this code just uses the same approaches as all the code written for the `Home` screen. Some points of interest though:

- Added a `Slider` component that is built on `slick-carousel`
- Added a `requestAll` approach for retrieving data from the network, that leverages `Promise.all` and iterates through all the provided items making a HTTP request for each. This allows us to retrieve multiple `opening_hours` in one go for example
- Like the `Home` screen, each element of the `Place` screen has been abstracted out into its own component to increase modularity and show what constitutes the screen at a glance. It also defers the responsibility of the data handling to the specific component
- Opening hours are mapped to the days of the week in a JS opinionated way: 0 = Sunday, 6 = Saturday, See `client/src/containers/Place/data/placeSelectors.js` lines 3 and 64 which are responsible for the mapping and sorting of the opening hours

## To test
- cd into `client` dir and run `npm install` to install the `Slider` dependencies
- In the admin, make sure that each Place you want to see on a Place screen has:
  - At least one `image`
  - At least one `opening_hour`
  - At least one `USP`
  - At least one `vital_info`
 - Go to http://localhost:3000/ and select a Place that meets the above criteria
 - You should see the Place screen intro, which should show a Background Image, Title, Subtitle, and an Arrow
 - Clicking on the arrow should scroll you down to the main content
 - You should see the name of the Place, an Image `Slider`, Vital Info, Reasons to Visit, Contact info and Opening hours!
 - You should see an 'x' button in the top-right of the screen, clicking this should return you to the Home screen, and you can continue using the app to look at other cities, places, filters etc.

## Reference screenshots
### Place screen `Intro`
![image](https://user-images.githubusercontent.com/3612297/125727518-9bc85792-d30f-42a6-b6a6-ef84f024ca55.png)

### Place screen content
![image](https://user-images.githubusercontent.com/3612297/125727549-31b87595-d322-4b40-ba04-3d77372f3c54.png)
